samtranslator/model/eventsources/pull.py

```diff
@@ -287,7 +287,26 @@ def get_policy_arn(self):  # type: ignore[no-untyped-def]
         return ArnGenerator.generate_aws_managed_policy_arn("service-role/AWSLambdaMSKExecutionRole")  # type: ignore[no-untyped-call]
 
     def get_policy_statements(self):  # type: ignore[no-untyped-def]
-        return None
+        document = None
+        if self.SourceAccessConfigurations:
+            for conf in self.SourceAccessConfigurations:
+                if isinstance(conf, dict) and conf.get("Type") == "CLIENT_CERTIFICATE_TLS_AUTH" and conf.get("URI"):
+                    document = {
```

Comment:
Shouldn't it automatically `return [document]` here?

Would also be good to include a comment explaining why we only pick one; is that a constraint of the underlying resource?

Reply:
Sorry, I'm not sure I understand your question. What do you mean by "explaining why we only pick one"? are you referring to the source access type? According to the [docs](https://docs.aws.amazon.com/lambda/latest/dg/API_SourceAccessConfiguration.html), CLIENT_CERTIFICATE_TLS_AUTH is the only one needed for mTLS auth.

Reply:
So if I have multiple `CLIENT_CERTIFICATE_TLS_AUTH` configurations, the `document` will only refer to the last `URI`, correct? As it loops over all of them, but each time it matches the condition, it overwrites `document`.

Reply:
Correct.

Reply:
Do you think we should end the loop after the first match?

Reply:
My question is: why? Why pick only one? Is it a constraint of the underlying resource?

Reply:
We only need one ARN of the secret key containing the client certificate. Can't think of why we would need more.

Reply:
It's not obvious to me reading the code why that assumption is made (without spending time reading through the internals and how it works). A "why" comment will help clarify for future readers.

Reply:
Perhaps another way to put it, if Lambda supports multiple `CLIENT_CERTIFICATE_TLS_AUTH`s in the same configuration, perhaps there's a valid use case that we shouldn't break. If Lambda doesn't support it, that is the "why".

Reply:
Ah I get what you mean now. I have not seen mentioned anywhere in the docs that Lambda supports multiple `CLIENT_CERTIFICATE_TLS_AUTH`. You're right. I can add a comment to make it clear for future readers.

Reply:
Added the comment.